src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -87,17 +94,17 @@ public static AppDomain CreateDomain(string friendlyName)
             throw new PlatformNotSupportedException();
         }
 
-        public int ExecuteAssembly(string assemblyFile) => ExecuteAssembly(assemblyFile, null, null, Configuration.Assemblies.AssemblyHashAlgorithm.None);
-        public int ExecuteAssembly(string assemblyFile, string[] args) => ExecuteAssembly(assemblyFile, args, null, Configuration.Assemblies.AssemblyHashAlgorithm.None);
-        public int ExecuteAssembly(string assemblyFile, string[] args, byte[] hashValue, Configuration.Assemblies.AssemblyHashAlgorithm hashAlgorithm)
+        public int ExecuteAssembly(string assemblyFile) => ExecuteAssembly(assemblyFile, null);
+        public int ExecuteAssembly(string assemblyFile, string[] args) 
         {
             if (assemblyFile == null)
                 throw new ArgumentNullException(nameof(assemblyFile));
-            /* TODO Assembly.LoadFrom is not implemented yet
-            Assembly assembly = Assembly.LoadFrom(path, hashValue, hashAlgorithm);
+            Assembly assembly = Assembly.LoadFrom(assemblyFile);
             return ExecuteAssembly(assembly, args);
-            */
-            return 0;
+        }
+        public int ExecuteAssembly(string assemblyFile, string[] args, byte[] hashValue, Configuration.Assemblies.AssemblyHashAlgorithm hashAlgorithm)
+        {
+            throw new PlatformNotSupportedException();
```

Comment:
Assemblies.LoadFrom overload api which takes assemblyHashAlgorithm also throws not supoorted exception because current AssemblyLoadContext implementation does not support checking the assemblyhash. Implementation of this would require non trivial implementation in VM and I believe the api is rarely used. I have discussed this with @jkotas and he agrees to punt this to later.